checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -471,7 +474,27 @@ public Void visitInstanceOf(InstanceOfTree tree, Void p) {
             // Handle them properly.
             return null;
         }
-        if (refTypeTree.getKind() == Tree.Kind.ANNOTATED_TYPE) {
+
+        PatternTree pattern = tree.getPattern();
+
+        if (pattern != null && pattern.getKind() == Tree.Kind.BINDING_PATTERN) {
+            BindingPatternTree bindingPattern = (BindingPatternTree) pattern;
+            VariableTree variable = bindingPattern.getVariable();
+            ModifiersTree modifiers = variable.getModifiers();
+
+            List<? extends AnnotationTree> annotationTrees = modifiers.getAnnotations();
+
+            List<? extends AnnotationMirror> annotations =
+                    TreeUtils.annotationsFromTypeAnnotationTrees(annotationTrees);
+
+            if (AnnotationUtils.containsSame(annotations, NULLABLE)) {
```

Comment:
Try to avoid the duplication with the code below.
Here, just extract the variable and then the below code should look whether the variable has modifiers. Or extract the annotations in two different ways, whichever is easier.
The error should only be raised in one place.